pkg/storage/store.go

```diff
@@ -52,10 +76,45 @@ func NewStore(
 		config.Database,
 		config.SSLMode,
 	)
-	db, err := sqlx.Connect("postgres", dataSourceName)
+	driver := "postgres"
+	if dbIamFlag {
+		config.Username = "app_user_iam"
+		if sess != nil {
+			// We want to get the credentials from the logged in AWS session rather than create directly,
+			// because the session conflates the environment, shared, and container metadata config
+			// within NewSession.  With stscreds, we use the Secure Token Service,
+			// to assume the given role (that has rds db connect permissions).
```

Comment:
Could you update this comment to explain why it's necessary to assume the dbIamRoleArn. This wasn't very intuitive and we ended up spending a lot of time troubleshooting it, in large part because the original comment wasn't very clear.